### ISSUE TITLE
Calculating Reg Loss 

### ISSUE THREAD

**Issue by waychin-weiqin** [fuy34/planerecover#10]
Hi, 

Just wondering in the script RecoverPlane.py, line 164, I assume you are trying to obtain the softmax of the non_plane prediction. Shouldn't it be `non_plane_mask = tf.exp(pred[:, :, :, -1:]) - tf.reduce_logsumexp(pred, axis=-1,keep_dims=True)` instead of just taking the difference between `pred[:, :, :, -1:]` and `tf.reduce_logsumexp(pred, axis=-1,keep_dims=True).`

**Comment by fuy34** [comment]
Hi, 
Your understanding is correct. 
Here the actual equation should be ```non_plane_mask = tf.log(tf.exp(pred[:, :, :, -1:])) - tf.reduce_logsumexp(pred, axis=-1,keep_dims=True)```. As ```tf.log(tf.exp(...))``` can be neutralized, we just put ```pred[:, :, :, -1:] - tf.reduce_logsumexp(pred, axis=-1,keep_dims=True)``` for simplicity.

**Comment by waychin-weiqin** [comment]
I see. Thanks for the clarification.
